extensions/eda/plugins/event_filter/normalize_keys.py

```diff
@@ -0,0 +1,50 @@
+"""
+normalize_keys.py:
+    An event filter that changes keys that contain non alpha numeric or
+    underscore to undersocres.
+    For instance, the key X-Y becomes the new key X_Y.
+    If there are consecutive non alpa numeric or under score, they would
+    be coalesced into a single underscore
+    For instance the key X/?Y becomes X_Y instead of X__Y
+
+    If there is a existing key with the normalized name, you can pass
+    in the overwrite argument to overwrite it with the new key
+Arguments:
+    * overwrite: Overwrite the values if there is a collision with a new key.
+
+Usage in a rulebook, a filter is usually attached to a source in the rulebook:
+    ....
+    sources:
+      - my_source:
+         ....
+        filters:
+          ansible.eda.normalize_keys:
+"""
+
+import multiprocessing as mp
+import re
+
+
+def main(event, overwrite=True):
+    logger = mp.get_logger()
+    logger.info("normalize_keys")
+    q = []
+    q.append(event)
+    regex = re.compile("[^0-9a-zA-Z_]+")
+    while q:
+        o = q.pop()
+        if isinstance(o, dict):
+            for key in list(o.keys()):
+                value = o[key]
+                q.append(value)
+                new_key = regex.sub("_", key)
+                if new_key != key:
+                    del o[key]
+                    if new_key in o and overwrite:
+                        o[new_key] = value
+                        logger.debug("Replacing %s with %s", key, new_key)
+                    elif new_key not in o:
+                        o[new_key] = value
+                        logger.debug("Replacing %s with %s", key, new_key)
+
```

Comment:
Probably have a log message indicating that the key is permanently removed if overwrite=false can be a good idea.